### PR TITLE
Add an array version of the benchmark  for Julia

### PR DESCRIPTION
**Pull request by alecloudenback**
Add a Julia array version that mimics the logic & flow of the pytorch example.

### CODE REVIEW

File: Julia/main.jl [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
     return Dict(
-            "mortality" => run_mortality_benchmarks(),
-            "exposures" => run_exposure_benchmarks(),
-            "basic_term_benchmark" => run_basic_term_benchmark(),
-            "savings_benchmark" => run_savings_benchmark(),
-        )
+        "mortality" => run_mortality_benchmarks(),
+        "exposures" => run_exposure_benchmarks(),
+        "basic_term_benchmark" => run_basic_term_benchmark(),
+        "basic_term_benchmark_array" => run_basicterm_array_benchmark(),
```
Comment (MatthewCaseres) [inline Julia/main.jl:17]: The README being generated right now has this yaml

```yaml 
basic_term_benchmark:
- Julia Benchmarks basic_term:
    minimum time: TrialEstimate(82.849 ms)
    result: 1.4489630534602132e7
- Python array numpy basic_term_m:
    minimum time: 78.52033000000347 milliseconds
    result: 14489630.534603368
basic_term_benchmark_array:
- 1.4489630534602132e7
```

to fix might involve modifying `run_basic_term_benchmark()` to put your new benchmark inside of that dictionary. basic_term_benchmark is the parent key for multiple dictionaries each representing a separate implementation

In CI the generate_readme step has a `cat readme` and this is how I checked the output from your branch. Can also run generate_readme.py for some quicker feedback.

**Review by MatthewCaseres — CHANGES_REQUESTED** [review]
ensure readme generation

File: Julia/main.jl (hunk 1 above)
Comment (alecloudenback) [reply Julia/main.jl:17]: should be fixed now

**Comment by alecloudenback** [comment]
Should be good to go now

**Comment by MatthewCaseres** [comment]
blazing fast, many thanks

**Pull request merged**